HMCLCore/src/main/java/org/jackhuang/hmcl/game/GameDumpCreator.java

```diff
@@ -0,0 +1,190 @@
+            printWriter.write("===== Minecraft JStack Dump =====\n");
+            for (Map.Entry<String, String> entry : infos.entrySet()) {
+                printWriter.write(entry.getKey());
+                printWriter.write(": ");
+
+                if (entry.getValue().contains("\n")) {
+                    // Multiple Line Value
+                    printWriter.write('{');
+                    printWriter.write('\n');
+
+                    String[] lines = entry.getValue().split("\n");
+                    for (int i = 0; i < lines.length; i++) {
+                        if (i != lines.length - 1) {
+                            printWriter.write("    ");
+                            printWriter.write(lines[i]);
+                            printWriter.write('\n');
+                        } else {
+                            // Last line
+                            if (lines[i].length() == 0) {
+                                // An empty last Line
+                                printWriter.write('}');
+                            } else {
+                                // Not an empty last lien
+                                printWriter.write("    ");
+                                printWriter.write(lines[i]);
+                                printWriter.write('\n');
+                                printWriter.write('}');
+                            }
+                        }
+                    }
+                } else {
+                    // Single Line Value
+                    printWriter.write(entry.getValue());
+                }
+                printWriter.write('\n');
+            }
+            printWriter.write('\n');
+            printWriter.write('\n');
+        }
+    }
+
+    private static final int TOOL_VERSION = 8;
+
+    private static final int DUMP_TIME = 3;
+
+    public static boolean checkDependencies() {
+        if (!CurrentJava.checkToolPackageDepdencies()) {
+            // Check whether tools.jar is available.
+            return false;
+        }
+        if (JavaVersion.CURRENT_JAVA.getParsedVersion() >= 9) {
+            // Method Process.pid() is provided on Java 9 or later.
+            // All the Operating System is accepted.
+            return true;
+        }
+        if (OperatingSystem.CURRENT_OS == OperatingSystem.WINDOWS) {
+            // On Windows, there is no ways to get the pid before Java 9 (We can only get the handle field of a Process).
+            return false;
+        } else if (OperatingSystem.CURRENT_OS == OperatingSystem.OSX || OperatingSystem.CURRENT_OS == OperatingSystem.LINUX) {
+            // On Linux or Mac, we can get field UnixProcess.pid field to get the pid even before Java 9.
+            return true;
+        }
+        // Unknown Operating System, no fallback available.
+        return false;
+    }
+
+    public static void writeDumpTo(long pid, File file) throws IOException, InterruptedException, ClassNotFoundException {
```

Comment:
```suggestion
    public static void writeDumpTo(long pid, Path file) throws IOException, InterruptedException, ClassNotFoundException {
```

能用 `Path` 就不要用 `File`。